### PR TITLE
Item OCP violation improvements

### PR DESCRIPTION
**Pull request by elsantner**
Introduces a system to dynamically register functionality by item class.
This eliminates almost all `instanceof` checks relating to `MediaItem` and `GroupItem`.

### CODE REVIEW

File: src/main/java/menagerie/gui/itemhandler/Items.java [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+/**
+ * Registry for implementations of item-type dependent functionality.
+ * Allows for easy-er extension of functionality than hard-coded instanceof checks.
+ * This does support inheritance in the expected way,
+ * i.e. if a subclass has more specific functionality registered, this will be used.
+ */
+public class Items {
+
+  // <Interface, <ItemType, Implementation>>
+  private static final HashMap<Class<?>, HashMap<Class<? extends Item>, Object>> register;
```
Comment (fragjulian) [inline src/main/java/menagerie/gui/itemhandler/Items.java]: [optional] rename to `registry`

File: src/main/java/menagerie/model/menagerie/importer/ImportJob.java [hunk 2]
```diff
@@ -329,7 +333,32 @@ private void trySimilar(Menagerie menagerie, MenagerieSettings settings) {
...
+          LOGGER.info("Found similar item (To ID: " + item.getId() + "): " + i.getId());
+          final var similarPair = new SimilarPair<>(item, (MediaItem) i, similarity);
+          similarTo.add(similarPair);
+        }
+      }
+    }
+    return anyMinimallySimilar;
+
+
+    /*boolean anyMinimallySimilar = false;
```
Comment (fragjulian) [inline src/main/java/menagerie/model/menagerie/importer/ImportJob.java]: leftover?

File: src/test/java/menagerie/model/menagerie/itemhandler/similarity/MediaItemSimilarityTest.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+import org.testfx.framework.junit5.ApplicationExtension;
+
+import java.io.File;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.lessThan;
+import static org.junit.jupiter.api.Assertions.*;
+
+@ExtendWith(ApplicationExtension.class)
+public class MediaItemSimilarityTest {
```
Comment (fragjulian) [inline src/test/java/menagerie/model/menagerie/itemhandler/similarity/MediaItemSimilarityTest.java:17]: nice coverage!

**Review by fragjulian — APPROVED** [review]
very well done! not only solves some SOLID violations but also reduces complexity sometimes

File: src/main/java/menagerie/gui/itemhandler/Items.java (hunk 1 above)
Comment (elsantner) [reply src/main/java/menagerie/gui/itemhandler/Items.java]: Thank you! I don't know why I didn't pick that name from the beginning...

File: src/main/java/menagerie/model/menagerie/importer/ImportJob.java (hunk 2 above)
Comment (elsantner) [reply src/main/java/menagerie/model/menagerie/importer/ImportJob.java]: Indeed, thank you!

**Pull request merged**